### ISSUE TITLE
Why does mlp.jl not use softmax?

### ISSUE THREAD

**Issue by Gregliest** [FluxML/model-zoo#287]
Why does this [line](https://github.com/FluxML/model-zoo/blob/master/vision/mnist/mlp.jl#L44) in mlp.jl not use a `softmax`? For the loss function, [logitcrossentropy](https://fluxml.ai/Flux.jl/stable/models/losses/#Flux.Losses.crossentropy), "the prediction ŷ is supposed to sum to one". I'm relatively new to ML: is this a bug, or am I misunderstanding something? 

Using `softmax` like `Dense(32, nclasses), softmax)` gives much worse results.

**Comment by darsnack** [comment]
That line refers to `crossentropy`. `logitcrossentropy` applies the `softmax` for you before computing the `crossentropy` loss.

**Comment by Gregliest** [comment]
Cool, thanks for the explanation!
